hack/lib/catalogsource.bash

```diff
@@ -156,6 +82,43 @@ EOF
   logger.success "CatalogSource installed successfully"
 }
 
+function build_image {
+  local name build_dir
+  name=${1:?Pass a name of image to be built as arg[1]}
+  build_dir=${2:?Pass a directory path for the build as arg[2]}
+
+  if ! oc get buildconfigs "$name" -n "$OLM_NAMESPACE" >/dev/null 2>&1; then
+    logger.info "Create an image build for ${name}"
+    oc -n "${OLM_NAMESPACE}" new-build --binary \
+      --strategy=docker --name "$name"
+  else
+    logger.info "${name} image build is already created"
+  fi
+
+  # Fetch previously created ConfigMap or remove empty file
+  oc -n "${OLM_NAMESPACE}" get configmap "${name}-sha1sums" \
+    -o jsonpath='{.data.'"$name"'\.sha1sum}' \
+    > "${rootdir}/_output/${name}.sha1sum" 2>/dev/null \
+    || rm -f "${rootdir}/_output/${name}.sha1sum"
+
+  if ! [ -f "${rootdir}/_output/${name}.sha1sum" ] || \
+      ! sha1sum --check --status "${rootdir}/_output/${name}.sha1sum"; then
+    logger.info 'Build the image in the cluster-internal registry.'
+    oc -n "${OLM_NAMESPACE}" start-build "${name}" \
+      --from-dir "${build_dir}" -F
+    mkdir -p "${rootdir}/_output"
+    find "${build_dir}" -type f -exec sha1sum {} + \
+      > "${rootdir}/_output/${name}.sha1sum"
+    oc -n "${OLM_NAMESPACE}" delete configmap "${name}-sha1sums" --ignore-not-found=true
+    oc -n "${OLM_NAMESPACE}" create configmap "${name}-sha1sums" \
+      --from-file="${rootdir}/_output/${name}.sha1sum"
+    rm -f "${rootdir}/_output/${name}.sha1sum"
+  else
+    logger.info "${name} build is up-to-date."
+  fi
+}
```

Comment:
we are using this approach because it's not possible to inject the previously built images using CI operator in a different image build context, right?

Reply:
Correct, I did not find a way to get the exact pullspec for the image so that it could be used in the following image build. The pull spec is only available in the "test" containers when running tests, through `dependencies` section of the ci-operator config. And did not get any answer for this on #forum-testplatform: https://coreos.slack.com/archives/CBN38N3MW/p1652275831765409